src/main/java/org/assertj/core/api/AbstractClassAssert.java

```diff
@@ -349,6 +349,45 @@ public SELF hasAnnotation(Class<? extends Annotation> annotation) {
     return myself;
   }
 
+  /**
+   * Verifies that the actual {@code Class} has the given class as direct superclass (as in {@link Class#getSuperclass()}).
+   * <p>
+   * Example:
+   * <pre><code class='java'> // this assertion succeeds as Object has no superclass:
+   * assertThat(Object.class).hasSuperclass(null);
```

Comment:
Minor `hasNoSuperclass` would be better than `hasSuperclass(null)` but in this PR.

Reply:
Done in #1752.